File: src/main/java/net/vulcanmc/vulcaneconomy/commands/Balance.java
package net.vulcanmc.vulcaneconomy.commands;

import net.vulcanmc.vulcaneconomy.VulcanEconomy;
import net.vulcanmc.vulcaneconomy.rest.Currency;
import net.vulcanmc.vulcaneconomy.rest.User;
import net.vulcanmc.vulcaneconomy.rest.Users;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.UUID;

public class Balance implements CommandExecutor {
    private final VulcanEconomy plugin;

    public Balance(VulcanEconomy plugin) {
        this.plugin = plugin; // Store the plugin in situations where you need it.
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        Player player = (Player) sender;
        Currency currency = new Currency();
        if(args.length == 1) {
            try {

                UUID uuid = VulcanEconomy.getPlugin().getUUIDIfExists(args[0]);
                if(uuid == null ) {
                    sender.sendMessage(ChatColor.RED + "Null profile lookup for " + args[0]);
                    return false;
                }

                if(Users.userExists(uuid) && Users.getUser(uuid).hasAccount(new Currency()))
                {
                    User user = Users.getUser(uuid);
                        Long balance = user.getAccount(currency).getBalance();
                        sender.sendMessage(ChatColor.GOLD + args[0] + "'s Balance: " + new Currency().getSymbol() + balance);
                        return true;
                } else {
                    sender.sendMessage(ChatColor.RED + "User " + args[0] + " does not have an account!");
                    return false;
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        } else {
            if (Users.userExists(player.getUniqueId())) {
                if (Users.getUser(player).hasAccount(currency)) {
                    Long balance = Users.getUser(player).getAccount(currency).getBalance();
                    sender.sendMessage(ChatColor.GOLD + "Balance: " + new Currency().getSymbol() + balance);
                    return true;
                } else {
                    sender.sendMessage(ChatColor.RED + "User " + player.getName() + " does not have an account!");
                    return false;
                }
            } else {
                sender.sendMessage(ChatColor.RED + "User " + player.getName() + " does not exist!");
                return false;
            }
        }
        return false;
    }
}

File: src/main/java/net/vulcanmc/vulcaneconomy/commands/Balance.kt
package net.vulcanmc.vulcaneconomy.commands

import net.vulcanmc.vulcaneconomy.VulcanEconomy
import net.vulcanmc.vulcaneconomy.rest.Currency
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class Balance(private val plugin: VulcanEconomy)// Store the plugin in situations where you need it.
    : CommandExecutor {

    override fun onCommand(sender: CommandSender, cmd: Command, label: String, args: Array<String>): Boolean {


        val defCurrency = plugin.currencies.defaultCurrency;

         var posCounter = 0;
        var useCache = true;
        if(args.size>=1 && (args[0].equals("false") )){
            useCache=false;
            posCounter++;
        }
        if (args.size >= (posCounter+1)) {
            val username = args[posCounter]
            val uuid = Bukkit.getOfflinePlayer(username).uniqueId
            val acc = plugin.accounts!!.getAccount(uuid, defCurrency)
            val currencies: ArrayList<Currency> = plugin.currencies.currencies
            for(currency in currencies) {
                val acc = plugin.accounts!!.getAccount(uuid, currency)
                if (currency.id.equals(plugin.currencies.defaultCurrency.id)) {
                    sender.sendMessage(plugin.prefix + "Main Balance: " + acc?.getBalance(useCache)?.toLong() + " "+ acc?.currency?.symbol)

                } else {
                    sender.sendMessage(plugin.prefix + acc?.getBalance(useCache)?.toLong() + " "+ acc?.currency?.symbol)

                }
            }
        } else if (sender is Player && args.size < (posCounter+1)) {

            val acc = plugin.accounts!!.getAccount(sender, defCurrency)


            val currencies: ArrayList<Currency> = plugin.currencies.currencies
            for(currency in currencies) {
                val acc = plugin.accounts!!.getAccount(sender, currency)

                if (currency.id.equals(plugin.currencies.defaultCurrency.id)) {
                    sender.sendMessage(plugin.prefix + "Main Balance: " + acc?.getBalance(useCache)?.toLong() + " "+ acc?.currency?.symbol)

                } else {
                    sender.sendMessage(plugin.prefix + acc?.getBalance(useCache)?.toLong() + " "+ acc?.currency?.symbol)

                }
            }


        }
        return true
    }
}
